File: Solutions/4033-longest-subsequence-with-non-zero-bitwise-xor/solution.py
class Solution:
    def longestSubsequence(self, nums: List[int]) -> int:
        n = len(nums)
        all_zero = 1
        result = 0
        for i in nums:
            if i!=0:
                all_zero = 0
            result ^= i
        if result:
            return n
        if all_zero == 1:
            return 0
        return n-1

File: Solutions/4033-longest-subsequence-with-non-zero-bitwise-xor/solution.cpp
class Solution {
public:
    int longestSubsequence(vector<int>& nums) {
        int n = nums.size();
        if (n==0){
            return 0;
        }
        int result = nums[0];
        int flag = 0;
        for(int i=1; i<n; i++){
            result ^= nums[i];
            if(nums[i]!=0){
                flag = 1;
            }
        }
        if (result){
            return n;
        }
        if (flag){
            return n-1;
        }
        return 0;
        
    }
};
